piff/gp_interp.py

```diff
@@ -0,0 +1,324 @@
+# Copyright (c) 2016 by Mike Jarvis and the other collaborators on GitHub at
+# https://github.com/rmjarvis/Piff  All rights reserved.
+#
+# Piff is free software: Redistribution and use in source and binary forms
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+
+"""
+.. module:: sklearn_gp_interp
+"""
+
+from .interp import Interp
+from .star import Star, StarFit
+
+import numpy as np
+
+class GPInterp(Interp):
+    """
+    An interpolator that uses sklearn.gaussian_process to interpolate a single surface.
+
+    :param  attr_interp: A list of star attributes to interpolate from
+    :param  kernel:      A Kernel object indicating how the different stars are correlated.
+    :param  optimizer:   Optimizer to use for optimizing the kernel.  Set to `None` to skip
+                         kernel optimization.
```

Comment:
Explain what the default is.  Also, can we come up with a better name than 'fmin_l_bfgs_b'?  I guess that's sklearn's name for this, but it's highly non-intuitive, so if we can give a more legible alias, I think that would be helpful.